src/hashset.c

```diff
@@ -1429,16 +1423,16 @@ int hashsetNext(hashsetIterator *iter, void **elemptr) {
     while (1) {
         if (iter->index == -1 && iter->table == 0) {
             /* It's the first call to next. */
+            if (iter->hashset->tables[0] == NULL) {
+                /* empty hashset, we're done */
+                break;
+            }
             if (iter->safe) {
                 hashsetPauseRehashing(iter->hashset);
             } else {
                 iter->fingerprint = hashsetFingerprint(iter->hashset);
             }
             iter->index = 0;
-            /* skip the rehashed slots in table[0] */
-            if (hashsetIsRehashing(iter->hashset)) {
-                iter->index = iter->hashset->rehashIdx;
-            }
```

Comment:
Why can't we skip already rehashed indices in table 0 (the old table)? They have been moved to table 1. Probing chains don't matter here.

Reply:
The iterator increments normally, but rehash increments the reverse index, so they can't be compared. This is just the quick fix - we could convert iterators to use the same order as rehash and scan - then we'd be able to take shortcuts like this.

Reply:
I took a look and changing the iteration order is actually easy, so I did that (which means we can skip the buckets now)

Reply:
Oh, right, the iterator just does index++. I forgot that part. +_+